packages/apm/src/integrations/express.ts

```diff
@@ -56,42 +56,66 @@ export class Express implements Integration {
  * // error handler
  * app.use(function (err, req, res, next) { ... })
  */
-function wrap(fn: Function, getCurrentHub: () => Hub): RequestHandler | ErrorRequestHandler {
+function wrap(fn: Function): RequestHandler | ErrorRequestHandler {
   const arrity = fn.length;
 
   switch (arrity) {
     case 2: {
       return function(this: NodeJS.Global, _req: Request, res: Response): any {
-        const span = getCurrentHub().startSpan({
-          description: fn.name,
-          op: 'middleware',
-        });
-        res.once('finish', () => span.finish());
+        // tslint:disable: no-unsafe-any
+        const transaction = (res as any).getTransaction && (res as any).getTransaction();
+        if (transaction) {
+          const span = transaction.startChild({
+            description: fn.name,
+            op: 'middleware',
+          });
+          res.once('finish', () => {
+            span.finish();
+          });
+        }
+        // tslint:enable: no-unsafe-any
         return fn.apply(this, arguments);
       };
     }
     case 3: {
       return function(this: NodeJS.Global, req: Request, res: Response, next: NextFunction): any {
-        const span = getCurrentHub().startSpan({
-          description: fn.name,
-          op: 'middleware',
-        });
-        fn.call(this, req, res, function(this: NodeJS.Global): any {
-          span.finish();
-          return next.apply(this, arguments);
-        });
+        // tslint:disable: no-unsafe-any
+        const transaction = (res as any).getTransaction && (res as any).getTransaction();
+        if (transaction) {
+          const span = transaction.startChild({
+            description: fn.name,
+            op: 'middleware',
+          });
+          fn.call(this, req, res, function(this: NodeJS.Global): any {
+            span.finish();
+            return next.apply(this, arguments);
+          });
+        } else {
```

Comment:
It's not necessary to make this function call twice in separate branches. We do `fn.call` anyway, so we can just drop this `if (transaction)` dance inside:


```
const span = transaction && transaction.startChild({ ... });
fn.call(this, req, res, function(this: NodeJS.Global): any {
  if (span) span.finish();
  return next.apply(this, arguments);
});
```